openmc/mgxs/mgxs.py

```diff
@@ -6239,14 +6239,17 @@ def get_pandas_dataframe(self, groups='all', nuclides='all',
                 df = df[df['group out'].isin(groups)]
 
         mesh_str = 'mesh {0}'.format(self.domain.id)
+        surfaces = df[(mesh_str,'surf')]
+        df.drop(columns=[(mesh_str,'surf')],inplace=True)
+        df.insert(len(self.domain.dimension),(mesh_str,'surf'),surfaces)
```

Comment:
Use [pandas' shift](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.shift.html)

Reply:
Panda's shift requires knowing where the column is first, and then moving it where you want it. 
The issue here is that pandas creates the order arbitrarily. So I don't know where it shows up, and therefore I don't know how much to shift it or in what direction. 
So for the test results to match & pass, I had to impose that it's at the end. The best solution I found was to drop & insert it where I want. That way, I don't need to know where it was.

Reply:
Good point. Can you specify the column placement directly then ?
like df = df[[mesh_str, 'surf', other_column_name_1, other_column_name_2]]

Reply:
Approving the PR after that. @nelsonag @paulromano do you have more comments

Reply:
It's a little more complicated than that because the column mesh_str is a dataframe itself which contains columns x, y, z, surf. 
The sub-columns of that mesh column can't be set easily 'inplace' (that is, without a copy, delete, replace) since it is embedded in df. It might be possible, but I spent some hours on it without success. There wasn't much literature on embedded data frames.

Reply:
Is this piece of code going to be acceptable?